streaming/src/main/java/io/gearpump/streaming/javaapi/Processor.java

```diff
@@ -35,6 +40,29 @@ public Processor(Class<T> taskClass, int parallelism) {
     this._parallelism = parallelism;
   }
 
+  public Processor(DataSink dataSink, int parallelism, String description,  UserConfig taskConf, ActorSystem system) {
+    io.gearpump.streaming.Processor p = DataSinkProcessor.apply(dataSink, parallelism, description, taskConf, system);
+    this._taskClass = p.taskClass();
+    this._parallelism = p.parallelism();
+    this._description = p.description();
+    this._userConf = p.taskConf();
+  }
+
+  public Processor(DataSource source, int parallelism, String description,  UserConfig taskConf, ActorSystem system) {
+    io.gearpump.streaming.Processor p = DataSourceProcessor.apply(source, parallelism, description, taskConf, system);
```

Comment:
Do you mean call other constructor by this()?
It looks like JDK6 don't support that.